src/inmanta/server/services/environment_metrics_service.py

```diff
@@ -189,6 +192,36 @@ async def start(self) -> None:
             self.schedule(
                 self.flush_metrics, COLLECTION_INTERVAL_IN_SEC, initial_delay=COLLECTION_INTERVAL_IN_SEC, cancel_on_stop=True
             )
+            # Cleanup metrics once per hour
+            self.schedule(self._cleanup_old_metrics, interval=3600, initial_delay=0, cancel_on_stop=True)
+
+    async def _cleanup_old_metrics(self) -> None:
+        """
+        Clean up metrics that are older than the retention time specified in the environment_metrics_retention
+        environment setting.
+        """
+        async with Environment.get_connection() as con:
+            environments = await Environment.get_list(connection=con)
```

Comment:
Couldn't we do all this in SQL? Or don't you think it would be worth it?

Reply:
Good idea. For some reason I didn't think about writing this into a single query.